pkg/pillar/cmd/nim/nim.go

```diff
@@ -920,13 +920,15 @@ func tryDeviceConnectivityToCloud(ctx *devicenetwork.DeviceNetworkContext) bool
 		}
 	} else {
 		if rtf {
+			// The fact that cloud replied with a status code shows that the cloud is UP, but not functioning
```

Comment:
In this case does it make sense to leave Works at whatever it was, or should we set it to true?
The old code set if to false (which is suboptimal) but leaving it at false might also be a bad idea.

Reply:
Don't see a reason why it shouldn't be set to true. Will do that.

Reply:
done